rectools/models/vector.py

```diff
@@ -116,39 +132,43 @@ def _process_implicit_scores(
 
     def rank(
         self,
-        subject_ids: np.ndarray,
+        subject_ids: InternalIds,
         k: int,
-        ui_csr_for_filter: tp.Optional[sparse.csr_matrix],  # only relevant for u2i recos
-        sorted_item_ids_to_recommend: tp.Optional[np.ndarray],  # whitelist
+        filter_so_csr: tp.Optional[sparse.csr_matrix] = None,  # subect-object interactions, only relevant for u2i case
```

Comment:
`filter_so_csr` is  not clear. Seen objects? It is highly uncommon. We'd better stick to `ui_csr_for_filter` or `filter_ui_csr` of `filter_viewed_csr`. At least it can be read easily.

Reply:
Also i don't like changing names for same things within our own code. Now we have `sorted_object_whitelist` == `sorted_item_ids_to_recommend` in other places

Reply:
1. I don't want to use `users`, `items`, `ui` words here since this ranker doesn't operate with such concepts.
2. I think `whitelist` is clearer (because it's shorter and commonly used) than `item_ids_to_recommend`. I'm going to use this name everywhere later. And it will have even more sense when we add `blacklist`

Reply:
fixed as discussed (renamed to `filter_pairs_csr`)